lib/srv/authhandlers.go

```diff
@@ -273,9 +275,20 @@ func (h *AuthHandlers) UserKeyAuth(conn ssh.ConnMetadata, key ssh.PublicKey) (*s
 	}
 	teleportUser := cert.KeyId
 
+	connectionDiagnosticID := cert.Extensions[teleport.CertExtensionConnectionDiagnosticID]
+
 	// only failed attempts are logged right now
 	recordFailedLogin := func(err error) {
 		failedLoginCount.Inc()
+
+		if err := maybeAppendDiagnosticTrace(ctx, connectionDiagnosticID, h.c.AccessPoint,
```

Comment:
Can we make `maybeAppendDiagnosticTrace` a member of `AuthHandlers`? Then you won't have to pass it access point. Doesn't look like it's used anywhere else.